bundles/zig/zig_lexer.moon

```diff
@@ -0,0 +1,138 @@
+    'enum', 'errdefer', 'error', 'export', 'extern', 'fn', 'for', 'if',
+    'inline', 'noalias', 'orelse', 'or', 'packed', 'promise', 'pub', 'resume',
+    'return', 'linksection', 'struct', 'suspend', 'switch', 'test',
+    'threadlocal', 'try', 'union', 'unreachable', 'usingnamespace', 'var',
+    'volatile', 'while'
+  }
+
+  -- Special words
+  special = c 'special', word {
+      'true', 'false', 'null', 'undefined', '@addWithOverflow', '@alignCast',
+      '@alignOf', '@as', '@asyncCall', '@atomicLoad', '@atomicRmw', '@atomicStore',
+      '@bitCast', '@bitOffsetOf', '@bitReverse', '@bitSizeOf', '@boolToInt',
+      '@breakpoint', '@byteOffsetOf', '@byteSwap', '@call', '@cDefine',
+      '@ceil', '@cImport', '@cInclude', '@clz', '@cmpxchgStrong',
+      '@cmpxchgWeak', '@compileError', '@compileLog', '@cos', '@ctz',
+      '@cUndef', '@divExact', '@divFloor', '@divTrunc', '@embedFile',
+      '@enumToInt', '@errorName', '@errorReturnTrace', '@errorToInt',
+      '@errSetCast', '@exp2', '@export', '@exp', '@fabs', '@fence',
+      '@fieldParentPtr', '@field', '@floatCast', '@floatToInt', '@floor',
+      '@frameAddress', '@frameSize', '@frame', '@Frame', '@hasDecl',
+      '@hasField', '@import', '@intCast', '@intToEnum', '@intToError',
+      '@intToFloat', '@intToPtr', '@log10', '@log2', '@log', '@memcpy',
+      '@memset', '@mod', '@mulAdd', '@mulWithOverflow', '@OpaqueType',
+      '@panic', '@popCount', '@ptrCast', '@ptrToInt', '@rem', '@returnAddress',
+      '@round', '@setAlignStack', '@setCold', '@setEvalBranchQuota',
+      '@setFloatMode', '@setRuntimeSafety', '@shlExact', '@shlWithOverflow',
+      '@shrExact', '@shuffle', '@sin', '@sizeOf', '@splat', '@sqrt',
+      '@subWithOverflow', '@tagName', '@TagType', '@This', '@truncate',
+      '@trunc', '@typeInfo', '@typeName', '@TypeOf', '@Type', '@unionInit',
+      '@Vector'
+  }
+
+  -- Record definitions
+  struct_def = sequence {
+    c 'keyword', word { 'struct', 'enum', 'error', 'union' }
+    c 'whitespace', space^1
+    c 'type_def', ident
+  }
+
+  -- Function declarations
+  fdecl = sequence {
+    c 'keyword', 'fn'
+    c 'whitespace', space^1
+    c 'fdecl', ident
+  }
+
+  -- Primitive types
+  primitive = word {
+    'anyerror', 'bool', 'c_int', 'c_longdouble', 'c_longlong', 'c_long',
+    'comptime_float', 'comptime_int', 'c_short', 'c_uint', 'c_ulonglong',
+    'c_ulong', 'c_ushort', 'c_void', 'f128', 'f16', 'f32', 'f64', 'isize',
+    'noreturn', 'type', 'usize', 'void'
+  }
+  type = c 'type', any {primitive}
+  integer = P'i' * digit^1
+  integer_type = c 'type', integer
+  unsigned = P'u' * digit^1
+  unsigned_type = c 'type', unsigned
+
+  -- Types, library modules and constants
+  important = upper^1 * (upper + lower + digit + '_')^0
+  important_name = c 'constant', important
+
+  -- Identifiers
+  identifier = c 'identifier', ident
+
+  -- Operators
+  operator = c 'operator', S'+=%-*/<>&|^~?!:;,.()[]{}'
+
+  P {
```

Comment:
Just a FYI: since there's no actual grammar being used (the grammar table contains only the one 'all' entry) it would be sufficient to return just the `any {` block directly and not use `P` at all.

Reply:
Updated as suggested. Thanks!